README.md

```diff
@@ -117,47 +120,171 @@ If you'd like to get a message when the button is pressed or released, call the
+getting the initial state of the pin and turning on interrupts. Without it, 
+you could get the state of the pin, it could change states, and then you could 
+start waiting on it for interrupts. If that happened, you would be out of sync.
 
 ## SPI
 
-A SPI bus is a common multi-wire bus used to connect components on a circuit
-board. A clock line drives the timing of sending bits between components. Bits
-on the `MOSI` line go from the master (usually the processor running Linux) to
-the slave, and bits on the `MISO` line go the other direction. Bits transfer
-both directions simultaneously. However, much of the time, the protocol used
-across the SPI bus has a request followed by a response and in these cases, bits
-going the "wrong" direction are ignored.
-
-The following shows an example ADC that reads from either a temperature sensor
-on CH0 or a potentiometer on CH1.
+A Serial Peripheral Interface (SPI, pronounced "spy") bus is a common multi-wire 
+bus used to connect components on a circuit board. A clock line drives the 
+timing of sending bits between components. Bits on the Master Out Slave In `MOSI` 
+line go from the master (usually the processor running Linux) to the slave, 
+and bits on the Master In Slave Out `MISO` line go the other direction. Bits 
+transfer both directions simultaneously. However, much of the time, the 
+protocol used across the SPI bus has a request followed by a response and in 
+these cases, bits going the "wrong" direction are ignored.  This will become 
+more clear in the example below.
+
+The following shows an example MCP3008 Analog to Digital Converter (ADC) that reads from 
+an analog moisture sensor, converts the analog measurements to digital, and 
+sends the digital measurements to SPI pins on a Raspberry Pi.  An ADC is 
+used to convert the analog signal from a sensor to a digital signal, so the 
+signal can be read by the main processor running Linux (e.g. Raspberry Pi.) 
+A Raspberry Pi can't read an analog signal, so it needs an ADC to convert the
+signal to digital.
+
+TODO UPDATED SCHEMATIC?
 
 ![SPI schematic](assets/images/schematic-adc.png)
 
-The protocol for talking to the ADC is described in the [MCP3202
-Datasheet](http://www.microchip.com/wwwproducts/Devices.aspx?dDocName=en010532).
-Sending a 0x64 first reads the temperature and sending a 0x74 reads the
-potentiometer.
+The protocol for talking to the ADC can be found on the MCP3008 data sheet 
+which you will find on the [MCP3008 product page](http://www.microchip.com/wwwproducts/en/MCP3008).  If 
+you are new to hardware, a data sheet can be intimidating but they become
+much easier to read once you know what you are looking for.  The protocol
+is very similar to an application program interface (API) in the software world.
+
+In the case of the MCP3008, the protocol is found in section 5.0 "SERIAL 
+COMMUNICATION" on page 19.  The protocol will tell you the position and 
```

Comment:
Remove the page number since that could easily change in a spec sheet revision. It's less common for section numbers to change, and I think that's enough for someone reading this to find it.